lumicks/pylake/tests/test_file.py

```diff
@@ -283,11 +283,16 @@ def test_invalid_access(h5_file):
 def test_missing_metadata(h5_file_missing_meta):
     f = pylake.File.from_h5py(h5_file_missing_meta)
     if f.format_version == 2:
-        with pytest.warns(UserWarning) as record:
+        with pytest.warns(UserWarning) as records:
             scans = f.scans
             assert len(scans) == 1
-            assert len(record) == 1
-            assert record[0].message.args[0] == "Scan 'fast Y slow X no meta' is missing metadata and cannot be loaded"
+            assert np.any(
+                [
+                    record.message.args[0]
+                    == "Scan 'fast Y slow X no meta' is missing metadata and cannot be loaded"
+                    for record in records
+                ]
+            )
```

Comment:
what caused this change to be necessary? it wasn't clear to me from the numpy changlelog

Reply:
Yeah, I also didn't expect it initially, but the reason for this change is that you get multiple warnings when accessing that scan (including non-`UserWarnings`):
``` DeprecationWarning('Passing None into shape arguments as an alias for () is deprecated.'), category : 'DeprecationWarning', filename : 'C:\\Conda\\envs\\test_pylake_latest\\lib\\site-packages\\h5py\\_hl\\dataset.py', lineno : 767, line : None}
{message : DeprecationWarning('Passing None into shape arguments as an alias for () is deprecated.'), category : 'DeprecationWarning', filename : 'C:\\Conda\\envs\\test_pylake_latest\\lib\\site-packages\\h5py\\_hl\\dataset.py', lineno : 767, line : None}
{message : UserWarning("Scan 'fast Y slow X no meta' is missing metadata and cannot be loaded"), category : 'UserWarning', filename : 'D:\\widget_hacks\\pylake\\lumicks\\pylake\\file.py', lineno : 201, line : None}```

All we want to really test is whether our warning is there, so that's why I changed it. Anyways, I just had another look and seems like pytest has a nice API for actually checking whether the string matches too, so I updated it to use that.